stablehlo/dialect/StablehloOps.td

```diff
@@ -1440,10 +1440,10 @@ def StableHLO_AllToAllOp : StableHLO_Op<"all_to_all",
 
 def StableHLO_ReduceOp: StableHLO_ShapedInterfaceOp<"reduce", [
       RecursiveMemoryEffects,
-      SameVariadicOperandSize,
-      InferTensorTypeWithReify,
+      SameVariadicOperandSize /*reduce_c3*/,
```

Comment:
I don't know if this is a bug or a feature, but I have discovered that `SameVariadicOperandSize` doesn't seem to work. For example, the following snippet of code passes verification:
```mlir
func.func @main(%arg0: tensor<4x4xf32>, %arg1 : tensor<f32>) -> (tensor<4x4xf32>) {
  %0 = "stablehlo.reduce"(%arg0, %arg1, %arg1) ({
    ^bb0(%arg2: tensor<f32>, %arg3: tensor<f32> ):
      %1 = stablehlo.constant dense<1.0> : tensor<f32>
      "stablehlo.return"(%1) : (tensor<f32>) -> ()
  }) {
    dimensions = dense<[]> : tensor<0xi64>
  } : (tensor<4x4xf32>, tensor<f32>, tensor<f32>) -> tensor<4x4xf32>
  func.return %0: tensor<4x4xf32>
}
```

If I'm getting this right, this needs a fix and also raises the priority of #1033 and #1036.

Reply:
Ah okay looks like we have #1334 for this.